prow/cmd/gangway/main.go

```diff
@@ -184,6 +191,12 @@ func main() {
 	gangway.RegisterProwServer(grpcServer, &gw)
 	grpc_prometheus.Register(grpcServer)
 
+	// Create gRPC health check endpoint. This can be used by the ESPv2 proxy
+	// container if Gangway is deployed for Cloud Endpoints.
+	healthcheckGrpc := health.NewServer()
+	healthgrpc.RegisterHealthServer(grpcServer, healthcheckGrpc)
+	healthcheckGrpc.SetServingStatus(serviceNameForHealthCheckGrpc, healthpb.HealthCheckResponse_SERVING)
```

Comment:
Should this come after `reflection.Register(grpcServer)`?  Since this shouldn't be called until the grpc server is fully operational I would have expected this to come next to `healthHTTP.ServeReady()`, right before we begin listening and service requests.

Reply:
My understanding is that the health check will start to be served at the same time that the gangway rpc endpoints become available. IOW, this code block does nothing until the grpc service comes online (with our call to ListenAndServe below).

FWIW there is another example on GitHub that also follows this pattern of setting up the healthcheck server first and then calling reflection afterwards: https://github.com/apssouza22/grpc-production-go/blob/70f12e02f004bf3007b14ec342d14009aa2007cc/server/server.go#L97-L103

Reply:
I looked through the upstream code and that seems right :+1: 
It would be great to include a comment explaining that this doesn't actually start serving the health status until serve is called since this behaves differently from `healthHTTP.ServeReady()` below.

Reply:
Done.